src/espIdf/arduino/addArduinoComponent.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Project: ESP-IDF VSCode Extension
+ * File Created: Thursday, 2nd June 2020 10:57:18 am
+ * Copyright 2020 Espressif Systems (Shanghai) CO LTD
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { getEspIdfVersion } from "../../utils";
+import { spawn, ChildProcess } from "child_process";
+import * as idfConf from "../../idfConfiguration";
+import * as treeKill from "tree-kill";
+import { join } from "path";
+import { ensureDir } from "fs-extra";
+import { OutputChannel } from "../../logger/outputChannel";
+import { Logger } from "../../logger/logger";
+import { Progress } from "vscode";
+
+export class ArduinoComponentInstaller {
+  private readonly ARDUINO_ESP32_URL =
+    "https://github.com/espressif/arduino-esp32.git";
+  private readonly projectDir: string;
+  private cloneProcess: ChildProcess;
+  constructor(projectDir: string) {
+    this.projectDir = projectDir;
+  }
+
+  public cancel() {
+    if (this.cloneProcess && !this.cloneProcess.killed) {
+      treeKill(this.cloneProcess.pid, "SIGKILL");
+      this.cloneProcess = undefined;
+      OutputChannel.appendLine("\n❌ [Arduino ESP32 Cloning] : Stopped!\n");
+    }
+  }
+
+  public async cloneArduinoInComponentsFolder(branchToUse: string) {
+    const componentsDir = join(this.projectDir, "components");
+    await ensureDir(componentsDir);
+    return new Promise((resolve, reject) => {
+      this.cloneProcess = spawn(
+        `git`,
```

Comment:
Maybe call `checkGitExists()` function before this to ensure user had git installed and if not show the specific error message (just a precautionary step)

Reply:
`getEspIdfVersion` calls `checkGitExists`